test/preconditioner/isai_kernels.cpp

```diff
@@ -476,180 +464,177 @@ TEST_F(Isai, HipIsaiGeneratePartialExcessIsEquivalentToRef)
     auto a2 = a1;
     gko::kernels::reference::isai::generate_tri_inverse(
         ref, mtx.get(), inverse.get(), a1.get_data(), a2.get_data(), false);
-    gko::array<index_type> da1(hip, a1);
+    gko::array<index_type> da1(exec, a1);
     auto e_dim = a1.get_data()[num_rows];
     auto e_rhs = Dense::create(ref, gko::dim<2>(e_dim, 1));
     std::fill_n(e_rhs->get_values(), e_dim, 123456);
-    auto de_rhs = gko::clone(hip, e_rhs);
+    auto de_rhs = gko::clone(exec, e_rhs);
     // overwrite -1 values with inverse
     d_inverse->copy_from(lend(inverse));
 
     gko::kernels::reference::isai::scatter_excess_solution(
         ref, a1.get_const_data(), e_rhs.get(), inverse.get(), 0, num_rows);
-    gko::kernels::hip::isai::scatter_excess_solution(
-        hip, da1.get_const_data(), de_rhs.get(), d_inverse.get(), 0, num_rows);
+    gko::kernels::EXEC_NAMESPACE::isai::scatter_excess_solution(
+        exec, da1.get_const_data(), de_rhs.get(), d_inverse.get(), 0, num_rows);
 
     GKO_ASSERT_MTX_NEAR(inverse, d_inverse, 0);
     ASSERT_GT(e_dim, 0);
 }
 
 
-TEST_F(Isai, HipIsaiScatterExcessSolutionAIsEquivalentToRef)
+TEST_F(Isai, IsaiScatterExcessSolutionAIsEquivalentToRef)
 {
     initialize_data(matrix_type::general, 702, 45);
     const auto num_rows = mtx->get_size()[0];
     gko::array<index_type> a1(ref, num_rows + 1);
     auto a2 = a1;
     gko::kernels::reference::isai::generate_general_inverse(
         ref, mtx.get(), inverse.get(), a1.get_data(), a2.get_data(), false);
-    gko::array<index_type> da1(hip, a1);
+    gko::array<index_type> da1(exec, a1);
     auto e_dim = a1.get_data()[num_rows];
     auto e_rhs = Dense::create(ref, gko::dim<2>(e_dim, 1));
     std::fill_n(e_rhs->get_values(), e_dim, 123456);
-    auto de_rhs = gko::clone(hip, e_rhs);
+    auto de_rhs = gko::clone(exec, e_rhs);
     // overwrite -1 values with inverse
     d_inverse->copy_from(lend(inverse));
 
     gko::kernels::reference::isai::scatter_excess_solution(
         ref, a1.get_const_data(), e_rhs.get(), inverse.get(), 0, num_rows);
-    gko::kernels::hip::isai::scatter_excess_solution(
-        hip, da1.get_const_data(), de_rhs.get(), d_inverse.get(), 0, num_rows);
+    gko::kernels::EXEC_NAMESPACE::isai::scatter_excess_solution(
+        exec, da1.get_const_data(), de_rhs.get(), d_inverse.get(), 0, num_rows);
 
     GKO_ASSERT_MTX_NEAR(inverse, d_inverse, 0);
     ASSERT_GT(e_dim, 0);
 }
 
 
-TEST_F(Isai, HipIsaiScatterExcessSolutionSpdIsEquivalentToRef)
+TEST_F(Isai, IsaiScatterExcessSolutionSpdIsEquivalentToRef)
 {
     initialize_data(matrix_type::spd, 100, 64);
     const auto num_rows = mtx->get_size()[0];
     gko::array<index_type> a1(ref, num_rows + 1);
     auto a2 = a1;
     gko::kernels::reference::isai::generate_general_inverse(
-        ref, mtx.get(), inverse.get(), a1.get_data(), a2.get_data(), true);
-    gko::array<index_type> da1(hip, a1);
+        ref, mtx.get(), inverse.get(), a1.get_data(), a2.get_data(), false);
```

Comment:
the next excess solution also uses true. Should it be true or false?

Reply:
it should stay `true`, good catch!